### PR TITLE
Fix hparam_overrides bug

### LINKED ISSUE #1886: hparam_overrides not working
## 🐛 Bug

Error when using `hparam_overrides` which was recently introduced by: #1797. This is the log:
```
Traceback (most recent call last):
  File "main.py", line 211, in <module>
    model = RelationEmbeddingModelLit.load_from_checkpoint(
  File "/home/ubuntu/anaconda3/envs/py38/lib/python3.8/site-packages/pytorch_lightning/core/lightning.py", line 1601, in load_from_checkpoint
    update_hparams(hparams, hparam_overrides)
UnboundLocalError: local variable 'hparams' referenced before assignment
```

### To Reproduce

1. Load a checkpoint using the `hparam_overrides` argument:
```
MyPLModel.load_from_checkpoint(checkpoint_path, hparam_overrides={key: new_val})
```
### Expected behavior

No bug.

### Environment

* CUDA:
	- GPU:
		- Tesla V100-SXM2-16GB
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.1
	- pyTorch_debug:     False
	- pyTorch_version:   1.4.0
	- pytorch-lightning: 0.7.7-dev
	- tensorboard:       2.2.1
	- tqdm:              4.43.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.1
	- version:           #117-Ubuntu SMP Wed Apr 8 09:52:02 UTC 2020

### PR DESCRIPTION
**Pull request by artidoro**
## What does this PR do?
Fixes #1886

Please let me know if you think there should be a test. I have tested it locally and it has solve my problem.

## Did you have fun?
:)

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1598,7 +1598,7 @@ def __init__(self, hparams: dict):
 
         # override the hparam keys that were passed in
         if hparam_overrides is not None:
-            update_hparams(hparams, hparam_overrides)
+            update_hparams(checkpoint['hparams'], hparam_overrides)
```
Comment (justusschock) [inline pytorch_lightning/core/lightning.py:1601]: Your fix would theoretically work, but I prefer to just move the update function before 

```
# overwrite hparams by the given file
checkpoint['hparams'] = hparams
```

which would have the same effect but porbably is a bit cleaner

Comment (artidoro) [reply pytorch_lightning/core/lightning.py:1601]: Yes that makes sense. I'll update the PR.

Comment (Borda) [reply pytorch_lightning/core/lightning.py:1601]: mind add test for it?

Comment (artidoro) [reply pytorch_lightning/core/lightning.py:1601]: @justusschock 
Actually, the ability to overwrite `hparams` that were passed through a .csv/.yaml file could be useful. If I move this update before we load the `hparams` from file we loose that. 

You know better, so I'll still update the PR if this is not the desired functionality.

Comment (artidoro) [reply pytorch_lightning/core/lightning.py:1601]: I'll look into adding a test @Borda.

Comment (justusschock) [reply pytorch_lightning/core/lightning.py:1601]: you shouldn't move this before we load them but before you reinject them to the checkpoint :)

**Comment by artidoro** [comment]
I updated the PR according to @Borda  and @justusschock comments.
If you could take another look that would be great.

Something that came up was how to behave when there were no hparams in the checkpoint, but the `hparam_overrides` argument is provided. In that case I decided to update the hparams to what was provided through `hparams_overrides`. I think this behavior is desirable.

**Comment by williamFalcon** [comment]
I would wait until #1896 is done

**Comment by williamFalcon** [comment]
@artidoro mind rebasing so we can merge?

**Comment by artidoro** [comment]
Ok sounds good. I'll do that as soon as possible. Might not be until Tuesday though.

**Comment by Borda** [comment]
@artidoro mind align with updated master and get this done? :rabbit:

**Comment by artidoro** [comment]
Ok will do!

**Comment by artidoro** [comment]
With the changes introduced by #1886 this is not an issue anymore.

**Pull request closed without merging**